app/models/BaseModel.ts

```diff
@@ -49,7 +63,43 @@ export default class BaseModel {
     }
   };
 
+  /**
+   * Returns a plain object representation of the model
+   *
+   * @returns {Record<string, any>}
+   */
   toJS = (): Record<string, any> => {
-    return { ...this };
+    const fields = getFieldsForModel(this);
+    return pick(this, fields) || [];
```

Comment:
What kind of issues were you seeing with the previous serialization style such that you went with explicit field defs? was a bunch of cruft making it into the json?

Reply:
The main thing is that I wanted a centralized store of which fields on a model _should_ be tracked which this did not provide, decorators felt like a nice DX to provide for this